### DISCUSSION TITLE
How can I add my own trained Tortoise voices?

### DISCUSSION THREAD

**Discussion by jonstreeter** [rsxdalv/TTS-WebUI#193] [Q&A]
I have a separate Tortoise installation that I used to train a custom voice and I want to incorporate it into the webui - but I can't figure out which files I need to copy or where to put them. 

Any help would be greatly appreciated!

**Comment by rsxdalv** [comment]
There's a voices folder, the folder button should open it from the UI, but it might not work on non-Windows systems. If your voices are not just audio files?
In that case if you have an autoregressive.pth (checkpoint) file, you need to make a copy of the tortoise models and replace the autoregressive one.
There are some topics about it here, and you can try to gamble on Chatgpt making the topic clearer.

**Reply by jonstreeter** [reply]
I tried copying the .pth model from the training folder of my Tortoise installation into the .. tts-generation-webui\data\models\tortoise fodler.

That did not work.

**Reply by rsxdalv** [reply]
Each collection of models needs it's own folder, like:
robot_voice_2/

autoregressive.pth
And the other pth files from the tortoise cache folder.

The official tortoise repo should be upgraded to allow only autoregressive.pth to be replaced, right now it needs everything.

**Reply by jonstreeter** [reply]
I've not been able to figure this out - any other ideas?

Where would I put the robot_voice_2 folder?

what files from a default tortoise training would i neep to put in that folder?

are there any other files I would need to put in any other folders in the webui installation?

I've searched here in the discussions and I can't find any other similar topics with relvenat info.
